x-pack/test/functional/services/ml/settings_calendar.ts

```diff
@@ -9,71 +9,26 @@ import expect from '@kbn/expect';
 
 import { FtrProviderContext } from '../../ftr_provider_context';
 import { MlCommonUI } from './common_ui';
-import { Navigation } from './navigation';
-import { MlADJobTable } from './job_table';
 
 export function MachineLearningSettingsCalendarProvider(
   { getService }: FtrProviderContext,
-  mlCommonUI: MlCommonUI,
-  navigation: Navigation,
-  jobTable: MlADJobTable
+  mlCommonUI: MlCommonUI
 ) {
   const testSubjects = getService('testSubjects');
   const retry = getService('retry');
   const comboBox = getService('comboBox');
 
   return {
-    async checkForCalendar(jobNumbers: number[], checkForExists: boolean = true) {
-      for await (const jobNumber of jobNumbers)
-        await jobTable.withDetailsOpen(`test_calendar_ad_${jobNumber}`, async () => {
-          await testSubjects[checkForExists ? 'existOrFail' : 'missingOrFail'](
-            jobTable.detailsSelector(
-              `test_calendar_ad_${jobNumber}`,
-              'mlJobRowDetailsSection-calendars'
-            )
-          );
-        });
-    },
-
-    async assertConnectedJobs(calendarId: string, regexp: RegExp) {
-      // Assert that the Calendar Management view shows that the calendar is only connected to the jobs
-      // applied at creation
-      await this.assertCalendarRowJobs(calendarId, regexp);
-
-      await navigation.navigateToAnomalyDetection();
-
-      // withDetailsOpen will open up the details of the specific job, and then collapse the details section afterwards
-      await this.checkForCalendar([1, 2]);
-
-      await this.checkForCalendar([3, 4], false);
-
-      await testSubjects.click('mlMainTab settings');
-      await testSubjects.click('mlCalendarsMngButton');
-      await testSubjects.click('mlEditCalendarLink');
+    async assertCalendarRowJobs(calendarId: string, expectedConnectedJobs: string[]) {
+      await this.filterWithSearchString(calendarId, 1); // makes sure the table only has the one row we want to check
+      const calendarRow = (await this.parseCalendarTable())[0];
+      const actualJobs: string[] = calendarRow.jobs.replaceAll(',', '').split(' ');
 
-      await this.selectJobGroup('multi-metric');
-      await this.saveCalendar();
-
-      const groupsAfterAddingMultiMetricsVisibleText = await testSubjects.getVisibleText(
-        'mlCalendarListColumnJobs'
+      const expectedWithinActual = expectedConnectedJobs.every((expected) =>
+        actualJobs.some((actual) => actual === expected)
```

Comment:
Checking for a subset here is not ideal because additional (unexpected) job ids should fail the assertion, so we should go for an exact match. There are multiple ways to do that, e.g. by sorting the arrays before comparing them.